woocommerce/woocommerce
# Add table name methods to OrderUtil

### Changes proposed in this Pull Request:

Adds methods to determine the names of the database tables currently being used for orders and order meta, without needing to make a separate check first to see if HPOS is enabled and in use. This also allows devs to get the table names dynamically instead of having to either hardcode them or call the `OrdersTableDataStore` class directly, which is discouraged since it's in the `Internal` namespace. 

### How to test the changes in this Pull Request:

One way to test this is with wp-cli, by opening an interactive shell to call the new methods directly. Note that you'll have to close the shell and re-open it after you change the state of HPOS.

1. With HPOS disabled, try using the shell to run the two new methods:
    ```
    wp shell
    wp> \Automattic\WooCommerce\Utilities\OrderUtil::get_table_for_orders()
    => string(8) "wp_posts"
    wp> \Automattic\WooCommerce\Utilities\OrderUtil::get_table_for_order_meta()
    => string(11) "wp_postmeta"
    ```
2. Now [enable HPOS](https://github.com/woocommerce/woocommerce/wiki/High-Performance-Order-Storage-Upgrade-Recipe-Book#easy-way) and run the commands again:
    ```
    wp shell
    wp> \Automattic\WooCommerce\Utilities\OrderUtil::get_table_for_orders()
    => string(12) "wp_wc_orders"
    wp> \Automattic\WooCommerce\Utilities\OrderUtil::get_table_for_order_meta()
    => string(17) "wp_wc_orders_meta"
    ```
